src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -57,6 +57,62 @@ internal DateTime StartTimeCore
             }
         }
 
+        /// <summary>Gets arguments.</summary>
+        /// <param name="startInfo">The start info with which to start the process.</param>
+        private string[] GetArgs(ProcessStartInfo startInfo)
+        {
+            string[] argv;
+
+            if (startInfo.UseShellExecute)
+            {
+                argv = new string[3] { GetExecPath(), "-c", startInfo.FileName + " " + startInfo.Arguments};
+            }
+            else
+            {
+                argv = ParseArgv(startInfo);
+            }
+
+            return argv;
+        }
+
+        /// <summary>Gets execution path</summary>
+        private string GetExecPath()
+        {
+            string[] allowedProgramsToRun = new string[3] { "xdg-open", "gnome-open", "kfmclient" };
+            string pathToProgram = string.Empty;
+
+            foreach (var program in allowedProgramsToRun)
+            {
+                if (!string.IsNullOrEmpty(program))
+                {
+                    break;
+                }
+                pathToProgram = GetPathToProgram(program);
+            }
+
+            if (string.IsNullOrEmpty(pathToProgram))
+            {
+                //return "/bin/sh"; possibly to do what we were doing earlier if none of the above three programs are already installed.
+            }
+
+            return pathToProgram;
+        }
+
+        private string GetPathToProgram(string program)
+        {
+            var path = Environment.GetEnvironmentVariable("PATH");
```

Comment:
Please review the corefx guidelines for "var" usage - these don't fit.

Reply:
The guidelines are at https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md 😃

Reply:
oops I didnt realize you guys will get notified of my PR just yet :) I need to cleanup all this now

Reply:
When you create a PR, folks will notice fairly quickly even if they aren't assigned it. So generally you throw up a PR either when you think you're done, or when you need feedback on what you have so far 😄